services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/MicrometerBasedMongoDbDeviceRegistryMetrics.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.deviceregistry.mongodb;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.hono.service.DeviceRegistryMetrics;
+import org.eclipse.hono.service.metric.MicrometerBasedMetrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.mongo.MongoClient;
+
+/**
+ * Metrics for Mongo DB based Device Registry service.
+ */
+public class MicrometerBasedMongoDbDeviceRegistryMetrics extends MicrometerBasedMetrics
+        implements MongoDbDeviceRegistryMetrics {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MicrometerBasedMongoDbDeviceRegistryMetrics.class);
+
+    private final MongoClient mongoClient;
+    private final String collectionName;
+
+    private final AtomicInteger tenantsCount;
+
+    /**
+     * Create a new metrics instance for the Device Registry service.
+     *
+     * @param vertx The Vert.x instance to use.
+     * @param registry The meter registry to use.
+     * @param mongoClient The client to use for accessing the Mongo DB.
+     * @param collectionName Tenants collection name in Mongo DB.
+     *
+     * @throws NullPointerException if either parameter is {@code null}.
+     */
+    public MicrometerBasedMongoDbDeviceRegistryMetrics(final Vertx vertx, final MeterRegistry registry,
+            final MongoClient mongoClient, final String collectionName) {
```

Comment:
given that we already have a TenantDao, IMHO it would make sense to add a *count()* method there and simply inject the DAO here instead of tying this class to the MongoClient. WDYT?
I also think that we can create the Gauge here and then do not need the *registerInitialTenantsCount* method to be invoked by the TenantManagementService which in turn would not need to implement LifeCycle then. WDYT?

Reply:
Of course, now since we don't have blocking operations is ok to register the Gauge at constructor time, therefore we don't need the LifeCycle. 
As for the _count()_ in TenantDao - do we need to make span for this method? It's not called from the API and I don't think it'll have much value on it's own, but on the other hand we will not be consistent with the other methods in the class. WDYT?

Reply:
> As for the count() in TenantDao - do we need to make span for this method? It's not called from the API and I don't think it'll have much value on it's own, but on the other hand we will not be consistent with the other methods in the class. WDYT?

As with the other DAO methods, we would accept a SpanContext param to be passed in. FMPOV we could then say that we only create a (child-)span if the passed in SpanContext is not null. That way we can let the caller of the method decide ...